### PR TITLE
Start bias includes neighboring tiles for better effect

### PR DESCRIPTION
**Pull request by yairm210**
This also means we don't need a special coastal check, since it's caught by the regular check
IIRC there are nations in Original Civ that have 'mountain' start bias - if so this will also enable that :)

### CODE REVIEW

File: core/src/com/unciv/logic/GameStarter.kt [hunk 1]
```diff
@@ -436,13 +438,20 @@ object GameStarter {
...
+                startBias.equalsPlaceholderText("Avoid []") -> {
+                    val tileToAvoid = startBias.getPlaceholderParameters()[0]
+                    preferredTiles.filter { tile ->
+                        !tile.getTilesInDistance(1).any {
+                            it.matchesTerrainFilter(tileToAvoid)
+                        }
+                    }
                 }
-                startBias == Constants.coast -> preferredTiles.filter { it.isCoastalTile() }
                 startBias in tileMap.naturalWonders -> preferredTiles  // passthrough: already failed
```
Comment (xlenstra) [inline core/src/com/unciv/logic/GameStarter.kt:449]: (not specific to this PR, but this just caught my eye)
Doesn't `tileMap.naturalWonders` only contain the wonders that are actually placed on this map? If so, aren't those the exact same that would have been filtered before this `for`-loop, effectively making this dead code? I'd assume you want to check if they are a natural wonder, and if so, not do anything, but that would be `ruleSet.terrains[startBias]!!.type == TerrainType.NaturalWonder`, right?

Additionally, the code above this PR for determining the starting space next to natural wonders has a variable `wonderNeighbor` which would make it likely to contain the tiles adjacent to natural wonders. However, unless I read the code incorrectly, it only finds the exact tile the wonder is on

**Review by xlenstra — APPROVED** [review]

**Review by will-ca — APPROVED** [review]

**Comment by SomeTroglodyte** [comment]
Didn't @SimonCeder code something to make start biases much friendlier, with that region stuff?

**Comment by yairm210** [comment]
He did, but this is still the function that filters tiles based on nation biases

**Comment by SimonCeder** [comment]
I'll point out that this function is used for placing starts on pre-made maps that don't have specified start locations.
When playing on a generated map all the regions code is run and returns a map with already defined start locations for each player, so the placement in GameStarter is not used in that case.

**Pull request merged**